### PR TITLE
MINOR Enable the GitHub build by default

### PR DESCRIPTION
**Pull request by mumrah**
Per the discussion in the [dev mailing list](https://lists.apache.org/thread/n4lwht8my06ydbzjrj0wxkj841r9t4ms), we should enable the new GitHub Actions workflows for all PRs. 

This patch does just that, while leaving the `github-actions-opt-in` variable in place for future experiments. 

Also included is a script that committers can use for approving workflow runs en-masse. Based on my experience, opening several tabs and clicking "Approve and Run" is rather tedious. The script utilizes the `gh` CLI which must be installed.

### CODE REVIEW

**Comment by mumrah** [comment]
We might want to wait on #17106 (and any follow-ups) to better utilize the build cache before turning this on for all PRs.

**Comment by mumrah** [comment]
@chia7712 WDYT, should we go ahead with this one?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -87,7 +87,6 @@ jobs:
       fail-fast: false
       matrix:
         java: [ 17, 11 ]
-    if: ${{ inputs.github-actions-opt-in == 'true' }}
```
Comment (chia7712) [inline .github/workflows/build.yml:94]: Should we update "Annotate checkstyle errors"? Removing the `inputs.github-actions-opt-in` or fix the "truthiness"

https://github.com/apache/kafka/blob/trunk/.github/workflows/build.yml#L78

**Review by chia7712 — COMMENTED** [review]
> should we go ahead with this one?

What you are about to do, I'm gonna to enjoy it, very, very much :smile: 

> while leaving the github-actions-opt-in variable in place for future experiments.

Instead of dropping all usages of `github-actions-opt-in`, maybe we should change the default value of `github-actions-opt-in` to true.

**Comment by mumrah** [comment]
> maybe we should change the default value of github-actions-opt-in to true

If we do that, we could probably just remove the input parameter. As it stands, we don't have any use cases for it, so maybe we should do that?

**Comment by chia7712** [comment]
> so maybe we should do that?

+1, we can always add it back in future experiments

**Comment by mumrah** [comment]
@chia7712 lmk if there's anything remaining on this one. Otherwise lets merge it

**Review by chia7712 — APPROVED** [review]

**Pull request merged**